test/e2e/storage/openstack_stress.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = SIGDescribe("openstack cloud provider stress [Feature:openstack]", func() {
+	f := framework.NewDefaultFramework("os-stress")
+	var (
+		client        clientset.Interface
+		namespace     string
+		instances     int
+		iterations    int
+		err           error
+		scNames       = []string{storageclass1, storageclass2, storageclass3, storageclass4}
+		policyName    string
+		datastoreName string
+	)
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("openstack")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		Expect(nodeList.Items).NotTo(BeEmpty(), "Unable to find ready and schedulable Node")
+
+		instancesStr := os.Getenv("OS_STRESS_INSTANCES")
+		Expect(instancesStr).NotTo(BeEmpty(), "ENV OS_STRESS_INSTANCES is not set")
+		instances, err = strconv.Atoi(instancesStr)
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing OS-STRESS-INSTANCES")
+		Expect(instances <= volumesPerNode*len(nodeList.Items)).To(BeTrue(), fmt.Sprintf("Number of Instances should be less or equal: %v", volumesPerNode*len(nodeList.Items)))
```

Comment:
A more informative error msg would be:
`..., fmt.Sprintf("Number of Instances (%d) should not exceed number of volumes across all nodes (%d), instances,  volumesPerNode*len(nodeList.Items))`

Reply:
Thanks, I have effected this.